packages/ddp-server/livedata_server.js

```diff
@@ -1606,9 +1664,18 @@ Object.assign(Server.prototype, {
     }
   },
 
-  _removeSession: function (session) {
+  _removeSession: function (session, callback = () => {}) {
     var self = this;
-    self.sessions.delete(session.id);
+    const f = () => {
+      self.sessions.delete(session.id);
+      callback();
+    };
+    if (session._expectingDisconnect) {
+      return f();
+    }
+    session.messageQueue = [];
+    session._pendingRemoveFunction = f;
+    session._removeTimeoutHandle = Meteor.setTimeout(f, DISCONNECT_GRACE_PERIOD);
```

Comment:
disconnect grace period should be also an option in the package to be replaced if someone wants to customize.